core/include/traccc/utils/particle.hpp

```diff
@@ -41,7 +41,8 @@ particle_from_pdg_number(const int pdg_num) {
             return detray::pion_minus<scalar_t>();
     }
 
-    return detray::muon<scalar_t>();
+    // TODO: Replace with `detray::invalid` in the future
+    return detray::pdg_particle<scalar_t>(0, 0.f, 0.f);
```

Comment:
Do we expect to reach this "legitimately"? I wonder if we should "error out" more aggressively.

Since this code may be used in both host and device code (Though I doubt that device code actually calls this. Does it?), throwing an exception is not on the table. But maybe something else?